### PR TITLE
[CI] Don't generate PDF doc unless necessary

### LINKED ISSUE #1085: Don't generate PDF doc unless necessary
### What

- Generate doc only in case of releases. Not for single commits.
- Force the generation of documentation if the branch starts matches `doc/*`

### Why

- We expect to save ~1min for each build 
- Save some bandwidth (Latex packages takes >200Mb)

### PR DESCRIPTION
**Pull request by payetvin**
close #1085

### CODE REVIEW

**Comment by flomnes** [comment]
Content of **NSISOutput.log**
```
# Run command: "C:/Program Files (x86)/NSIS/makensis.exe"  "D:/a/Antares_Simulator/Antares_Simulator/_build/_CPack_Packages/win64/NSIS/project.nsi"
# Output:
Processing config: C:\Program Files (x86)\NSIS\nsisconf.nsh


Processing script file: "D:/a/Antares_Simulator/Antares_Simulator/_build/_CPack_Packages/win64/NSIS/project.nsi" (ACP)


File: "D:/a/Antares_Simulator/Antares_Simulator/_build\ui\simulator\resources\help\\antares-general-reference-guide.pdf" -> no files found.


Usage: File [/nonfatal] [/a] ([/r] [/x filespec [...]] filespec [...] |


   /oname=outfile one_file_only)


Error in macro File on macroline 3


Error in script "D:/a/Antares_Simulator/Antares_Simulator/_build/_CPack_Packages/win64/NSIS/project.nsi" on line 336 -- aborting creation process


```

File: .github/workflows/windows-vcpkg.yml [hunk 1]
```diff
@@ -191,10 +179,19 @@ jobs:
            rm -rf src/tests/resources/Antares_Simulator_Tests
            rm -rf src/tests/resources/NonRegTests
            cd _build
+           mkdir -p ui\simulator\resources\help\
+           touch antares-general-reference-guide.pdf system-map-editor-reference-guide.pdfi antares-examples-library.pdf
```
Comment (flomnes) [inline .github/workflows/windows-vcpkg.yml]:
```suggestion
           touch antares-general-reference-guide.pdf system-map-editor-reference-guide.pdf antares-examples-library.pdf
```

File: .github/workflows/centos7.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ env:
 jobs:
 
   doc_generation:
+    if: ${{ github.event_name == 'release' || contains(github.ref_name, 'doc/') }}
```
Comment (flomnes) [inline .github/workflows/centos7.yml:24]:
```suggestion
    if: ${{ github.event_name == 'release' || startsWith(github.ref_name, 'doc/') }}
```
Slightly more accurate

File: src/distrib/win32/NSIS.template.in [hunk 3]
```diff
@@ -252,16 +252,16 @@ LangString LngDesktopLink ${LANG_ENGLISH} "Desktop Shortcut"
...
 LangString LngResInstallMenuLink ${LANG_ENGLISH} "Creating a Menu Shortcut..."
-;LangString LngResInstallMenuLink ${LANG_FRENCH}  "Cr�ation d'un Raccourci dans le menu..."
+;LangString LngResInstallMenuLink ${LANG_FRENCH}  "Cr?ation d'un Raccourci dans le menu..."
 
 
 
 
 LangString LngInstallFull ${LANG_ENGLISH}  "Full (Recommended)"
-;LangString LngInstallFull ${LANG_FRENCH}   "Complet (Recommand�)"
+;LangString LngInstallFull ${LANG_FRENCH}   "Complet (Recommand?)"
```
Comment (flomnes) [inline src/distrib/win32/NSIS.template.in:264]: Revert this

File: src/distrib/win32/NSIS.template.in [hunk 4]
```diff
@@ -213,7 +213,7 @@ InstType "$(LngInstallLite)"
 
 
 LangString LngWelcome ${LANG_ENGLISH} "Welcome to the Antares_Simulator ${PRODUCT_VERSION} Setup Wizard"
-;LangString LngWelcome ${LANG_FRENCH} "Bienvenue � l'assistant d'installation d'Antares ${PRODUCT_VERSION}"
+;LangString LngWelcome ${LANG_FRENCH} "Bienvenue ? l'assistant d'installation d'Antares ${PRODUCT_VERSION}"
```
Comment (flomnes) [inline src/distrib/win32/NSIS.template.in:216]: Revert this

File: src/distrib/win32/NSIS.template.in [hunk 5]
```diff
@@ -64,7 +64,7 @@ VIAddVersionKey /LANG=${LANG_ENGLISH} "ProductName" "${PRODUCT_NAME} ${PRODUCT_A
 VIAddVersionKey /LANG=${LANG_ENGLISH} "Comments" ""
 VIAddVersionKey /LANG=${LANG_ENGLISH} "CompanyName" "${PRODUCT_PUBLISHER}"
 VIAddVersionKey /LANG=${LANG_ENGLISH} "LegalTrademarks" "Antares_Simulator is a trademark of RTE"
-VIAddVersionKey /LANG=${LANG_ENGLISH} "LegalCopyright" "� ${PRODUCT_PUBLISHER}"
+VIAddVersionKey /LANG=${LANG_ENGLISH} "LegalCopyright" "? ${PRODUCT_PUBLISHER}"
```
Comment (flomnes) [inline src/distrib/win32/NSIS.template.in:67]: Revert

**Review by guilpier-code — APPROVED** [review]

File: src/distrib/win32/sourcefiles.inc.nsh [hunk 6]
```diff
@@ -13,12 +13,12 @@ File  ${ANTARES_SRC_DIR}\..\INSTALL.md
 File  ${ANTARES_SRC_DIR}\..\README.md
 
 ${SetOutPath} "$INSTDIR\Sources\docs"
-File /r  ${ANTARES_SRC_DIR}\..\docs\*.*
+File /nonfatal /r  ${ANTARES_SRC_DIR}\..\docs\*.*
 
 ${SetOutPath} "$INSTDIR\Sources\resources"
-File /r  ${ANTARES_SRC_DIR}\..\resources\*.*
+File /nonfatal /r  ${ANTARES_SRC_DIR}\..\resources\*.*
```
Comment (flomnes) [inline src/distrib/win32/sourcefiles.inc.nsh:19]: Revert this

**Review by flomnes — CHANGES_REQUESTED** [review]
1. Revert changes related to accents & encoding
2. At the top of each file, introduce env variable `GENERATE_DOC`
```yaml
GENERATE_DOC: ${{ github.event_name == 'release' || startsWith(github.ref_name, 'doc/') }}
```
3. Revert changes in sourcefiles.inc.nsh

**Review by flomnes — APPROVED** [review]

**Pull request merged**